### PR TITLE
refactor: remove task category from compute engine

### PR DESCRIPTION
**Pull request by guilhem-barthes**
## Description

Do not use task category in the compute engine

## How has this been tested?
Local E2E and CI 

## Checklist

- [x] [changelog](../CHANGELOG.md) was updated with notable changes
- [ ] documentation was updated

### CODE REVIEW

File: backend/substrapp/compute_tasks/volumes.py [hunk 1]
```diff
@@ -25,16 +25,15 @@ def get_volumes(ctx: Context):
...
-        # predicttuple and Xtraintuples
-        volume_mounts.extend(
-            [
-                _create_mount(dirs.task_dir, TaskDirName.Chainkeys),
-                _create_mount(dirs.task_dir, TaskDirName.OutModels),
-            ]
+
+    if ctx.has_output_of_kind(orchestrator.AssetKind.ASSET_MODEL):
+        volume_mounts.append(
+            _create_mount(dirs.task_dir, TaskDirName.OutModels),
```
Comment (guilhem-barthes) [inline backend/substrapp/compute_tasks/volumes.py:36]: Please someone confirms that we don't need more than checking `has_chainkeys` and outputs of specific types.

Comment (mblottiere) [reply backend/substrapp/compute_tasks/volumes.py:36]: Yes 👍 chainkeys are independent from task categories, volume mount should depend only on outputs.

**Review by mblottiere — APPROVED** [review]
lgtm, can you add a changelog entry please?

**Pull request merged**